frontend/ui/widget/touchmenu.lua

```diff
@@ -744,6 +746,105 @@ function TouchMenu:updateItems()
+            local menu_entry = val[i].text or (val[i].text_func and  val[i].text_func())
+            local next_text = menu_entry and (text .. "→ " .. tostring(menu_entry)) or text -- drop sub_item_table
+            TouchMenu:_recurse(val[i], path .. "." .. i, next_text, search_for, depth)
+        elseif i == "text_func" then
+            menu_text = v()
+        elseif i == "text" then
+            menu_text = v
+        end
+
+        if menu_text and Utf8Proc.lowercase(menu_text):find(search_for) then
+            table.insert(TouchMenu.foundMenuItem, {menu_text, path .. "." .. i, text})
+        end
+    end
+end
+
+function TouchMenu:onMenuSearch(search_for)
+    TouchMenu.foundMenuItem = {}
+    for i = 1, #self.tab_item_table do
+        TouchMenu:_recurse(self.tab_item_table[i], i, self.tab_item_table[i].text or "ROOT"..i, search_for, 0)
+    end
+
+--[[
+    for i = 1, #TouchMenu.foundMenuItem do
+        print("xxxxxxx->", i,
+            TouchMenu.foundMenuItem[i][1],
+            TouchMenu.foundMenuItem[i][2],
+            TouchMenu.foundMenuItem[i][3])
+    end
+]]
+end
+
+function TouchMenu:onOpenMenu(path, animate_time_s)
+    local function animate_func()
+        if not animate_time_s or animate_time_s == 0.0 then return end
+        local sleep_us = animate_time_s * 1e6
+        UIManager:forceRePaint()
+        ffiUtil.usleep(sleep_us)
+        UIManager:setDirty(nil, "ui")
+    end
+    -- first switch to correct MenuTab
+    local sep_pos = path:find("%.")
+    local tab_num = tonumber(path:sub(1, sep_pos - 1))
+    path = path:sub(sep_pos + 1)
+    local item = self.tab_item_table[tab_num]
+
+    if not tab_num then
+        return
+    end
+    self:switchMenuTab(tab_num)
+    self.bar:switchToTab(tab_num)
+
+    self:updateItems()
+
+    -- Now open the menus all the way down
+    local dummy, num_of_sep = path:gsub("%.", "%.")
+    while num_of_sep > 1 do
+        sep_pos = path:find("%.")
+        local identifier = path:sub(1, sep_pos -1)
+        local item_num = tonumber(identifier)
+        path = path:sub(sep_pos + 1)
+        if item_num then
+            animate_func()
+            item = item[item_num]
+            self:onMenuSelect(item)
+        elseif identifier == "sub_item_table" then
+            item = item.sub_item_table
+        end
+        num_of_sep = num_of_sep - 1
+    end
+
+    -- Now we are in the right menu, but maybe in the wrong page
+    sep_pos = path:find("%.")
+    local identifier = path:sub(1, sep_pos -1)
+    local item_num = tonumber(identifier)
+
+    if item_num then -- might be nil on a sub-menu-entry
+        local desired_page_num = math.floor(item_num / self.perpage)
+        if desired_page_num > 0 then
+            for i = 1, desired_page_num do
+                animate_func()
+                self:onNextPage()
+            end
+        end
+    end
+
+    -- clean up
+    TouchMenu.foundMenuItem = {}
+end
+
 function TouchMenu:switchMenuTab(tab_num)
```

Comment:
Any reason to put these 3 new TouchMenu methods here, in the middle of all the others related to the normal use of TouchMenu ?
Feels best to add them at the bottom, and mention than these next ones are related to menu search only.